gulp-core-build-typescript/src/TypeScriptConfiguration.ts

```diff
@@ -48,9 +48,27 @@ export class TypeScriptConfiguration {
     const oldConfig: ITsConfigFile<ts.Settings> = this.getGulpTypescriptOptions(buildConfig);
     const newConfig: ITsConfigFile<typescript.CompilerOptions> = oldConfig as any;
 
-    newConfig.compilerOptions.moduleResolution =
-      oldConfig.compilerOptions.moduleResolution === 'node' ?
-        typescript.ModuleResolutionKind.NodeJs : typescript.ModuleResolutionKind.Classic;
+    delete newConfig.compilerOptions.moduleResolution;
+
+    if (typeof oldConfig.compilerOptions.target === 'string') {
+      const target: string =
+        (oldConfig.compilerOptions.target as string).toUpperCase();
+      newConfig.compilerOptions.target = typescript.ScriptTarget[target];
```

Comment:
Can you add a note explaining that we're mapping a string into an enum integer here?  It's pretty nonobvious